app/main/helpers/frameworks/validation.py

```diff
@@ -0,0 +1,178 @@
+from functools import reduce
+from operator import add
+import re
+import six
+from werkzeug.datastructures import ImmutableOrderedMultiDict
+
+EMAIL_REGEX = r'^[^@^\s]+@[^@^\.^\s]+(\.[^@^\.^\s]+)+$'
+
+G7_CHARACTER_LIMIT = 5000
+
+G7_OPTIONAL_FIELDS = set([
+    "SQ1-1p-i", "SQ1-1p-ii", "SQ1-1p-iii", "SQ1-1p-iv",
+    "SQ1-1q-i", "SQ1-1q-ii", "SQ1-1q-iii", "SQ1-1q-iv", "SQ1-1cii", "SQ1-1i-ii",
+    "SQ1-1j-i", "SQ1-1j-ii", "SQ4-1c", "SQ3-1k", "SQ1-1i-i"
+])
+
+G7_EMAIL_VALIDATION_FIELDS = set(['SQ1-1o', 'SQ1-2b'])
+
+validators = {}
+
+
+def get_validator(framework):
+    """
+    Retrieves a validator by name, otherwise returns EmptyValidator.
+    """
+    if framework is not None:
+        framework_name = framework.get('slug', None)
+        return validators.get(framework_name, EmptyValidator)
+    else:
+        return EmptyValidator
+
+
+class FrameworkValidator(object):
```

Comment:
This is only used for framework declarations, so `FrameworkValidator` is a bit ambiguous.`FrameworkDeclarationValidator` or just `DeclarationValidator`?